discovery-provider/src/queries/get_notifications.py

```diff
@@ -402,3 +434,25 @@ def get_notifications(session: Session, args: GetNotificationArgs):
         for notification in notifications
     ]
     return notifications_and_actions
+
+
+class GetUnreadNotificationCount(TypedDict):
+    user_id: int
+    timestamp: Optional[datetime]
+    valid_types: Optional[List[NotificationType]]
+
+
+def get_unread_notification_count(session: Session, args: GetUnreadNotificationCount):
+    args["valid_types"] = args.get("valid_types", []) + default_valid_types  # type: ignore
+    resultproxy = session.execute(
+        unread_notification_count_sql,
+        {
+            "user_id": args["user_id"],
+            "valid_types": args.get("valid_types", None),
+            "timestamp_offset": args.get("timestamp", None),
+        },
+    )
+    unread_count = 0
+    for rowproxy in resultproxy:
+        unread_count = rowproxy[0]
```

Comment:
isn't this just setting unread_count to the last value in resultproxy? why not just do `unread_count = resultproxy[-1][0]`?

Reply:
you can't index into the resultproxy